### DISCUSSION TITLE
Longer [Computing Jacobian................] time than before

### DISCUSSION THREAD

**Discussion by Bearunner** [idaholab/moose#27627] [Q&A Modules: General] [answered]
Dear all, 

I use the [Porous flow] module to simulate thermal-hydraulic processes in porous media.

However, using the same inputs (input files and meshes, etc.) we created last year, the updated MOOSE version needs longer [Computing Jacobian................] time than the old version. 

**[old version MOOSE]: faster [Computing Jacobian................]**
Framework Information:
MOOSE Version:           git commit 2d9ac4d977 on 2023-03-29
LibMesh Version:         
PETSc Version:           3.16.6
SLEPc Version:           3.16.2
Current Time:            Wed Sep 13 00:25:15 2023
Executable Timestamp:    **Thu May 25 16:15:28 2023**

**[Newversion MOOSE]: slower [Computing Jacobian................]**
Framework Information:
MOOSE Version:           git commit 0b297bbb93 on 2024-02-15
LibMesh Version:
PETSc Version:           3.20.3
SLEPc Version:           3.20.1
Current Time:            Wed May 15 14:47:54 2024
Executable Timestamp:    **Tue May 14 19:25:11 2024**

Here is the log of the current version.

Time Step 1, time = 1, dt = 1

 0 Nonlinear |R| = 1.579338e+00

    Computing Jacobian................
      Computing DiracKernels                                                             **[  9.30 s]** [  488 MB]
    Finished Computing Jacobian                                                          **[ 97.48 s]** [  499 MB]

      0 Linear |R| = 1.579338e+00
      1 Linear |R| = 1.186822e-01
      2 Linear |R| = 9.897281e-02
      3 Linear |R| = 9.131680e-02
      4 Linear |R| = 2.607287e-02
      5 Linear |R| = 1.666023e-02
      6 Linear |R| = 1.267173e-02
      7 Linear |R| = 1.101330e-02
      8 Linear |R| = 3.424740e-03
      9 Linear |R| = 1.840177e-03
     10 Linear |R| = 9.638534e-05
     11 Linear |R| = 3.917202e-05
     12 Linear |R| = 1.628395e-05
     13 Linear |R| = 1.828763e-06
     14 Linear |R| = 1.241097e-06
     15 Linear |R| = 2.503577e-07
     16 Linear |R| = 2.143965e-08
     17 Linear |R| = 3.366161e-10
     18 Linear |R| = 1.418001e-11
     19 Linear |R| = 3.183701e-13

 1 Nonlinear |R| = 9.684284e-04
    Computing Jacobian.................                                                  **[ 96.20 s]** [  755 MB]
      0 Linear |R| = 9.684284e-04
      1 Linear |R| = 2.467006e-06
      2 Linear |R| = 2.153255e-06
      3 Linear |R| = 1.582451e-06
      4 Linear |R| = 2.866890e-07
      5 Linear |R| = 2.420947e-07
      6 Linear |R| = 2.013037e-07
      7 Linear |R| = 1.693336e-07
      8 Linear |R| = 1.184412e-08
      9 Linear |R| = 5.044444e-09
     10 Linear |R| = 1.785820e-09
     11 Linear |R| = 7.671236e-10
     12 Linear |R| = 2.124358e-10
     13 Linear |R| = 1.320051e-11
     14 Linear |R| = 5.053740e-12
     15 Linear |R| = 2.289863e-12
     16 Linear |R| = 2.860599e-13
     17 Linear |R| = 5.967287e-15
     18 Linear |R| = 1.015925e-16

 2 Nonlinear |R| = 3.515348e-11

 Solve Converged!

  Finished Solving                                                                       **[207.61 s]** [  539 MB]

Outlier Variable Residual Norms:
  porepressure: 2.350344e-11
  temperature: 2.573907e-11

Could you please help me with this? Thanks!

J

**Comment by GiudGiud** [comment, marked as answer]
Hello

We cant know which change over a year may have slowed it down. It could be anything.
I would recommend profiling
https://mooseframework.inl.gov/application_development/profiling.html

and trying to make the current version faster by proposing changes in the hot spots of the code
Guillaume
